portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -26,21 +27,60 @@
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
-    private ArrayList<String> messages = new ArrayList<>(Arrays.asList("Message 1", "Message 2", "Message 3"));
-
-    /**
-     * Convert ArrayList as JSON using GSON library.
-    */
-    private String convertArrayListToJson(ArrayList<String> messages) {
-        Gson gson = new Gson();
-        String json = gson.toJson(messages);
-        return json;
-    }
+    private ArrayList<Comment> comments = new ArrayList<>();
 
     @Override
     public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-        response.setContentType("text/html;");
-        // response.getWriter().println("<h1>Hello Ariana!</h1>");
-        response.getWriter().println(convertArrayListToJson(messages));
+        response.setContentType("application/json");
+        String json = new Gson().toJson(comments);
+        response.getWriter().println(json);
+        System.out.println("doGet:");
+        System.out.println(json);
     }
+
+    @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        // Get input from the form
+        String name = getNameField(request);
+        String location = getLocationField(request);
+        String content = getContentField(request);
+        // TODO check which field this pops up
+        if (name == null || location == null || content == null) {
+            response.setContentType("text/html");
+            response.getWriter().println("Please fill in all fields.");
+            return;
+        }
```

Comment:
This is fine for now.

For a production system, it would be more friendly if this validation happened in the browser, so there is quicker feedback. In fact, you will often see that web site forms will not enable the Submit button until the form passes validation. It is also good to indicate which field are required as well. But it also good to not overdo it on the required fields, as that tends to annoy users.